freedomofpress/securedrop-client
# Add action to (re-)generate and export conversation transcript and downloaded files

## Description

The shortest path to providing an option to export a entire conversation to journalists.
Entire conversation = freshly (re-)generated conversation transcript + all the downloaded files [1]

  [1] _On one hand, there is an option to download all files, which is easy to trigger. On the other hand, if a file was intentionally not downloaded, conditioning the export to an automatic download seems counter-productive._

## Context

This is a direct continuation of #1624, the same context applies.

## Implementation

- Renames the existing `export.Dialog` to `export.FileDialog` because that's what it is, and there is a parallel with the `export.PrintFileDialog`. breaking the symmetry would become confusing I think.
- Uses the `conversation/export.Dialog` name to export the entire conversation.
- Takes the existing _export dialog_ and _export transcript dialog_ as a reference
- Follows the existing patterns for testing the new dialog

## Test plan

1. Start the app
2. Select a conversation
3. For good measure download no file, or a file, or two (or all of them)
4. Trigger "Export Conversation" in the _conversation menu_ (a.k.a source menu, kebab menu...)
- [ ] Confirm that the dialog says it will export `4 files` if three files were downloaded (the transcript is the fourth file)
- [ ] Confirm the transcript is exported as expected
- [ ] Confirm the files are exported as expected
- [ ] Confirm that error messages are shown in the same ways as in the existing export dialog

(This [patch](https://github.com/freedomofpress/securedrop-client/commit/202f3d7dd9a075c4b8f1429bea5a18cb30c40483) allows to test this PR in a lightweight way. That being said, a proper export to a USB device is desirable.)
